### DISCUSSION TITLE
Can share screen with OBS Studio, but not with Firefox nor Chromium

### DISCUSSION THREAD

**Discussion by ghost** [hyprwm/Hyprland#1088] [Q&A]
I **can** share the screen with OBS Studio, but not with Firefox nor Chromium. They share a completely black screen.

I went through the [Hyprland screensharing wiki page](https://wiki.hyprland.org/Useful-Utilities/Screen-Sharing/), the [gist by PowerBall253](https://gist.github.com/PowerBall253/2dea6ddf6974ba4e5d26c3139ffb7580), andthe [official troubleshooting guide](https://github.com/emersion/xdg-desktop-portal-wlr/wiki/%22It-doesn't-work%22-Troubleshooting-Checklist).

All looks nice.

Except that the official troubleshooting guide says thatFirefox and Chromium point to Pipewire 0.2 while I have Pipewire 0.3. COuld this be it? They say I have to install a "Pipewire compat library", but I cannot find it in the Arch repos.

If it works with OBS Studio, I don't understand while it should notwork with Firefox or Chromium.

I have the NVIDIA proprietary drivers and yes, I'm starting Hyprland as the wiki tells me to start it in case of NVIDIA proprietary drivers, with all the environmental variables set.

**Comment by vaxerski** [comment]
verify foorefox is running in native wayland. For chroomoom, you also need to enable webrtc pipewire streaming (as well as launching in native wayland)

> Firefox and Chromium point to Pipewire 0.2

Ze what

**Comment by ghost** [comment]
> verify foorefox is running in native wayland

that was part of the links I posted. It does

>  For chroomoom, you also need to enable webrtc pipewire streaming (as well as launching in native wayland)

as above

> Ze what

the guide says that pipewire 0.2 clients are incompatible with pipewire 0.3 servers and need a "compat" library whatever that is.

In the meantime I found [this](https://gitlab.freedesktop.org/pipewire/pipewire/-/issues/2835) very related issue

**Reply by vaxerski** [reply]
then it seems like a pw issue :-/

**Reply by ghost** [reply]
No it doesn't. OBS is using pipewire too.

**Reply by vaxerski** [reply]
Read the issue that you linked, literally.

**Comment by ghost** [comment]
I will write to xdg-desktop-portal-wlr.

In the meantime, on the AUR there is a package called [hyprland-nvidia-git](https://aur.archlinux.org/packages/hyprland-nvidia-git), which contains a patch for screensharing on NVIDIA proprietary drivers. 

Where does this patch come from?

Where is the discussion that lead to this patch?

I cannot find relevant issue about screensharing here.

Maybe the one who came up with the patch can help me.
